src/TelemetryChannels/ServerTelemetryChannel/Shared/TelemetryChannelBuilderExtensions.cs

```diff
@@ -20,5 +20,41 @@ public static void UseSampling(this TelemetryProcessorChainBuilder builder, doub
         {
             builder.Use((next) => new SamplingTelemetryProcessor(next) { SamplingPercentage = samplingPercentage });
         }
+
+        /// <summary>
+        /// Adds <see cref="AdaptiveSamplingTelemetryProcessor"/> to the <see cref="TelemetryProcessorChainBuilder" />
+        /// </summary>
+        /// <param name="builder">Instance of <see cref="TelemetryProcessorChainBuilder"/></param>
+        /// <return>Instance of <see cref="TelemetryProcessorChainBuilder"/>.</return>
+        public static TelemetryProcessorChainBuilder UseAdaptiveSampling(this TelemetryProcessorChainBuilder builder)
+        {
+            return builder.Use((next) => new AdaptiveSamplingTelemetryProcessor(next));
+        }
+
+        /// <summary>
+        /// Adds <see cref="AdaptiveSamplingTelemetryProcessor"/> to the <see cref="TelemetryProcessorChainBuilder" />
+        /// </summary>
+        /// <param name="builder">Instance of <see cref="TelemetryProcessorChainBuilder"/></param>
+        /// <param name="maxTelemetryItemsPerSecond">Maximum number of telemetry items to be generated on this application instance.</param>
+        /// <return>Instance of <see cref="TelemetryProcessorChainBuilder"/>.</return>
+        public static TelemetryProcessorChainBuilder UseAdaptiveSampling(this TelemetryProcessorChainBuilder builder, double maxTelemetryItemsPerSecond)
+        {
+            return builder.Use((next) => new AdaptiveSamplingTelemetryProcessor(next) { MaxTelemetryItemsPerSecond = maxTelemetryItemsPerSecond });
+        }
+
+        /// <summary>
+        /// Adds <see cref="AdaptiveSamplingTelemetryProcessor"/> to the <see cref="TelemetryProcessorChainBuilder" />
+        /// </summary>
+        /// <param name="builder">Instance of <see cref="TelemetryProcessorChainBuilder"/></param>
+        /// <param name="settings">Set of settings applicable to dynamic sampling percentage algorithm.</param>
+        /// <param name="callback">Callback invoked every time sampling percentage evaluation occurs.</param>
+        /// <return>Instance of <see cref="TelemetryProcessorChainBuilder"/>.</return>
+        public static TelemetryProcessorChainBuilder UseAdaptiveSampling(
+            this TelemetryProcessorChainBuilder builder, 
+            SamplingPercentageEstimatorSettings settings,
```

Comment:
I'm concerned that this reduces discoverability... UseAdaptiveSampling(settings) is easily discoverable. Discovering a class and a way to "use" it is much much harder.